target/linux/rockchip/image/armv8.mk

```diff
@@ -12,6 +12,16 @@ define Device/friendlyarm_nanopi-r2s
 endef
 TARGET_DEVICES += friendlyarm_nanopi-r2s
 
+define Device/friendlyarm_nanopi-r4s
+  DEVICE_VENDOR := FriendlyARM
+  DEVICE_MODEL := NanoPi R4S
+  SOC := rk3399
+  UBOOT_DEVICE_NAME := nanopi-r4s-rk3399
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script nanopi-r4s | pine64-img | gzip | append-metadata
+  DEVICE_PACKAGES := kmod-r8169
+endef
+TARGET_DEVICES += friendlyarm_nanopi-r4s
```

Comment:
AFAICT the only difference between the 1G and 4G models is the required bootloader. Still, can you add a "4GB DDR" DEVICE_VARIANT here?

I'm still unsure, how we can differentiate between the devices at runtime to enforce usage of the correct image. But this page can be looked at when the 1G rev is implemented.

Reply:
> Still, can you add a "4GB DDR" DEVICE_VARIANT here?

Done.

> how we can differentiate between the devices at runtime to enforce usage of the correct image

FriendlyELEC made some changes to do that, and here is the source code:
https://github.com/friendlyarm/uboot-rockchip/commits/nanopi4-v2020.10

Reply:
This was not what i was referring to. For me, it was about the support-list of the image metadata which would be identical for both boards, despite the images can not be used interchangable.

Modifying the U-Boot might also be possible, however this should be done upstream as I'd like to avoid maintaining such a hack downstream.